### PR TITLE
Network get version info

### LINKED ISSUE #6185: getVersionInfo - networkAdmin Service

### PR DESCRIPTION
**Pull request by kimbor**
Implement NetworkGetVersionInfo query

**Related issue(s)**:

Fixes #6185 

**Notes for reviewer**:
Richard is working on a separate PR that makes changes to the way that version info is instantiated. There will be changes to SemanticVersion and related classes based on that work. Therefore, the version info related code is just stubs and I didn't bother writing tests for it since it will be replaced soon anyway.

### CODE REVIEW

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/config/SemanticVersionConfig.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package com.hedera.node.app.service.networkadmin.impl.config;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.swirlds.config.api.ConfigData;
+import com.swirlds.config.api.ConfigProperty;
+
+@ConfigData("version")
+public record SemanticVersionConfig(
+        @ConfigProperty(defaultValue = "0.39.0") SemanticVersion servicesVersion,
+        @ConfigProperty(defaultValue = "0.39.0") SemanticVersion hapiVersion) {}
```
Comment (rbair23) [inline hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/config/SemanticVersionConfig.java:26]: In my variant of this class, I defined the "servicesVersion" field to be called "services" and the hapiVersion to be "hapi". This was so in the config it would be "version.services" and "version.hapi" instead of "version.servicesVersion" and "version.hapiVersion" since this seemed redundant.

I agonized a bit over it though, because the existing fields are something more like "proto.version" and "services.version", implying they might be part of different config objects (with a @ConfigData of services or a @ConfigData of hapi).

I also agonized over what, if any, default version to give them. Ideally they would have no default version, so that if a build somehow failed to get a proper version, then the node would simply fail to start. Getting the version number right is essential for selecting which schema to use. If we do give it a default, I recommend we give it a default version of 0, which is the genesis version. If the node software starts but with no version information, then it will fail to start unless it is also a genesis case. If you tried to start the node software where the existing state had a higher version number, it would fail to start. So it seems like the safest default would be "0.0.0"

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java [hunk 2]
```diff
@@ -31,4 +32,20 @@ public interface NetworkInfo {
...
+    @NonNull
+    SemanticVersion servicesVersion();
+
+    /**
+     * Get the current deployed HAPI version running on the network.
+     *
+     * @return The HAPI version
+     */
+    @NonNull
+    SemanticVersion hapiVersion();
```
Comment (rbair23) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:50]: If we're using config to get the version information (as I see is the case in NetworkGetVersionINfoHandler) then I guess we don't need this yet?

File: hedera-node/hedera-network-admin-service-impl/build.gradle.kts [hunk 3]
```diff
@@ -25,6 +25,7 @@ dependencies {
     testImplementation(project(":hedera-node:node-app-service-token"))
     testImplementation(testFixtures(project(":hedera-node:node-config")))
     testImplementation(testFixtures(project(":hedera-node:node-app-spi")))
+    testImplementation(testFixtures(project(":hedera-node:node-app-service-mono")))
```
Comment (rbair23) [inline hedera-node/hedera-network-admin-service-impl/build.gradle.kts:28]: Why is this needed? Can the query handler be implemented entirely based on the modular code paths? Why does it need the mono service test fixtures?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java [hunk 4]
```diff
@@ -16,16 +16,29 @@
...
 import com.hedera.node.app.spi.info.NetworkInfo;
 import com.hedera.pbj.runtime.io.buffer.Bytes;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.regex.Pattern;
 
 /**
  * Implementation of {@link NetworkInfo} that delegates to the mono-service.
  */
 public class MonoNetworkInfo implements NetworkInfo {
+    private static final int MAX_VERSION_LENGTH = 100;
```
[the remark this answers is not in the document]
Comment (kimbor) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java]: Once I removed changes to NetworkInfo, I was also able to remove changes to this implementing class.

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/config/SemanticVersionConfig.java (hunk 1 above)
Comment (kimbor) [reply hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/config/SemanticVersionConfig.java]: A default is required in order for unit tests to work. I changed the default to "0.0.0" and I changed to use the naming you suggest.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java (hunk 2 above)
Comment (kimbor) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java]: Yep. Deleted changes.

File: hedera-node/hedera-network-admin-service-impl/build.gradle.kts (hunk 3 above)
Comment (kimbor) [reply hedera-node/hedera-network-admin-service-impl/build.gradle.kts]: I was able to do some refactoring and remove this dependency.

**Review by rbair23 — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by jeromy-cannon — APPROVED** [review]
